src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/ShipOutOfLuck.java

```diff
@@ -0,0 +1,156 @@
+public class ShipOutOfLuck extends SimpleHack<ShipOutOfLuckConfig> implements Listener {
+
+	public ShipOutOfLuck(SimpleAdminHacks plugin, ShipOutOfLuckConfig config) {
+		super(plugin, config);
+	}
+
+	@Override
+	public String status() {
+		StringBuilder builder = new StringBuilder(getClass().getSimpleName());
+		builder.append(" is ").append(isEnabled() ? "enabled" : "disabled").append(".").append("\n");
+		Set<Material> scuttleList = this.config.getScuttleList();
+		if (scuttleList.isEmpty()) {
+			builder.append("No scuttle blocks.");
+		}
+		else {
+			builder.append(" Scuttle blocks:\n");
+			builder.append(scuttleList.stream()
+					.map(effect -> "  • " + effect.name())
+					.collect(Collectors.joining("\n")));
+		}
+		return builder.toString();
+	}
+
+	// ------------------------------------------------------------
+	// Listeners
+	// ------------------------------------------------------------
+
+	@Override
+	public void registerListeners() {
+		this.plugin().registerListener(this);
+	}
+
+	@Override
+	public void unregisterListeners() {
+		HandlerList.unregisterAll(this);
+	}
+
+	@EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
+	public void preventBoatPlacements(PlayerInteractEvent event) {
+		switch (event.getAction()) {
+			case RIGHT_CLICK_AIR:
+			case RIGHT_CLICK_BLOCK:
+				break;
+			default:
+				return;
+		}
+		Block placedOn = event.getClickedBlock();
+		ItemStack placed = event.getItem();
+		if (!BlockAPI.isValidBlock(placedOn) || !this.config.isScuttleBlock(placedOn.getType())
+				|| !ItemAPI.isValidItem(placed) || !Tag.ITEMS_BOATS.isTagged(placed.getType())) {
+			return;
+		}
+		event.setCancelled(true);
+		plugin().debug("Prevented boat placement on [" + placedOn.getType().name() + "] by [" +
+				event.getPlayer().getName() + "] at [" + placedOn.getLocation() + "]");
+		//return;
+	}
+
+	@EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
+	public void preventBoatUsage(VehicleMoveEvent event) {
+		Vehicle vehicle = event.getVehicle();
+		if (vehicle.getType() != EntityType.BOAT) {
+			return;
+		}
+		List<Entity> passengers = vehicle.getPassengers();
+		if (passengers.isEmpty()) {
+			return;
+		}
+		Location destination = event.getTo();
+		if (event.getVehicle().isOnGround()) {
+			// If the boat is on top of a block, then we need to look below the boat
+			destination.add(0, -1, 0);
+		}
+		if (destination.getY() < 0 || destination.getY() >= destination.getWorld().getMaxHeight()) {
+			return;
+		}
+		Location[] locations = new Location[] {
+				destination,
+				destination.clone().add(-1, 0, 1),
+				destination.clone().add(0, 0, 1),
+				destination.clone().add(1, 0, 1),
+				destination.clone().add(-1, 0, 0),
+				destination.clone().add(1, 0, 0),
+				destination.clone().add(-1, 0, -1),
+				destination.clone().add(0, 0, -1),
+				destination.clone().add(1, 0, -1)
+		};
+		for (Location currentLocation : locations) {
+			if (!WorldAPI.isBlockLoaded(currentLocation)) {
```

Comment:
Those aren't free either performance wise and seem super unnecessary if we know a player is within 2 blocks of it. If those were _actually_ unloaded, there'd be a million other problems anyway.